app/hooks/useFreeze.ts

```diff
@@ -0,0 +1,13 @@
+import { useState } from "react";
+
+export const useFreeze = <FrozenType>(
+  initializer: FrozenType | (() => FrozenType)
+): FrozenType => {
+  const [frozenValue] = useState<FrozenType>(initializer);
+
+  return frozenValue;
+};
+
+export const useFreezeFunction = <FnType extends (...args: any[]) => any>(
+  fn: FnType
+) => useFreeze(() => fn);
```

Comment:
A couple of hooks added in I commonly use (although never in ts lol). Probably overkill but nice to have IMO

Reply:
> Note: these aren't actually used, but could well be useful in a future PR - should I wait and add them when needed?

I don't really know these hooks but I generally go with YAGNI. 
Not fussed, whatever ya'll think is best

Reply:
Had to push another change anyway so removed lol